### PR TITLE
Fix optional pinocchio dependency

### PR DESCRIPTION
**Pull request by gergondet**
Prior to this PR if ndcurves is built without pinocchio support a `find_package(pinocchio QUIET)` still finds its way into `ndcurvesConfig.cmake` which results in failure when one tries to `find_package(ndcurves REQUIRED)` in a downstream project.

This PR only adds pinocchio as a dependency when the `CURVES_WITH_PINOCCHIO_SUPPORT` is `ON`.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -29,12 +29,13 @@ CHECK_MINIMAL_CXX_STANDARD(11 ENFORCE)
 
 # Project dependencies
 ADD_PROJECT_DEPENDENCY(Eigen3 REQUIRED PKG_CONFIG_REQUIRES eigen3)
-ADD_PROJECT_DEPENDENCY(pinocchio QUIET)
+FIND_PACKAGE(pinnochio QUIET)
```
Comment (wxmerkt) [inline CMakeLists.txt]:
```suggestion
FIND_PACKAGE(pinocchio QUIET)
```

Comment (gergondet) [reply CMakeLists.txt]: My bad! I've fixed it now..

**Comment by gergondet** [comment]
Sorry to bump this but could it be merged now? If so would it be possible to tag a new release as well?

Thank you and happy new year!

**Comment by nim65s** [comment]
Happy new year !
Thanks for the reminder, I'll make a release in the day

**Pull request merged**